library/ecdsa.c

```diff
@@ -239,7 +239,7 @@ static int derive_mpi( const mbedtls_ecp_group *grp, mbedtls_mpi *x,
         MBEDTLS_MPI_CHK( mbedtls_mpi_shift_r( x, use_size * 8 - grp->nbits ) );
 
     /* While at it, reduce modulo N */
-    if( mbedtls_mpi_cmp_mpi( x, &grp->N ) >= 0 )
+    if( mbedtls_mpi_cmp_mpi_ct( x, &grp->N ) >= 0 )
```

Comment:
While this is a small improvement over the existing code, it leaves a larger leak open: the if branch itself. I think it would be better to always compute the result of the subtraction and then do a safe conditional swap (as above).